config/areas/users/dialogs.php

```diff
@@ -336,4 +336,69 @@
 		'submit'  => $fields['file']['submit']
 	],
 
+	// user disable TOTP
+	'user.totp.disable' => [
+		'pattern' => 'users/(:any)/totp/disable',
+		'load' => function (string $id) {
+			$user        = Find::user($id);
+			$currentUser = App::instance()->user();
+			$dialog      = [
+				'props' => [
+					'submitButton' => [
+						'text'  => I18n::translate('disable'),
+						'icon'  => 'protected',
+						'theme' => 'negative'
+					],
+				]
+			];
+
+			if (
+				$currentUser->isAdmin() === true &&
+				$currentUser->is($user) === false
+			) {
+				$dialog['component'] = 'k-remove-dialog';
+				$dialog['props']['text'] = I18n::template('login.totp.disable.admin', ['user' => $user->name()->or($user->email())]);
+			} else {
+				$dialog['component'] = 'k-form-dialog';
+				$dialog['props']['fields'] = [
+					'password' => [
+						'type'     => 'password',
+						'required' => true,
+						'counter'  => false,
+						'label'    => I18n::translate('login.totp.disable.label'),
+						'help'     => I18n::translate('login.totp.disable.help'),
+					]
+				];
+			}
+
+			return $dialog;
+		},
+		'submit' => function (string $id) {
+			$user     = Find::user($id);
+			$kirby    = App::instance();
+			$password = $kirby->request()->get('password');
+
+			try {
+				// Admins can disable TOTP without password
+				if (
+					$kirby->user()->isAdmin() === false ||
+					$kirby->user()->is($user) === true
+				) {
+					$user->validatePassword($password);
+				}
+
+				// Remove the TOTP secret from the account
+				$user->changeTotp(null);
+
+				return [
+					'message' => I18n::translate('login.totp.disable.success')
+				];
+			} catch (Exception $e) {
```

Comment:
I think we should only catch the specific exception we need to rethrow. All other exceptions can be passed upwards 1:1.

Reply:
You mean just catching `InvalidArgumentException`? This will still catch several exceptions as only one is specifically sending a 401 code that we'll need to prevent

Reply:
Yes, but IMO still better than catching every exception type. Especially because we re-throw a `InvalidArgumentException`, so if another exception type was to be thrown, the exception class would be converted incorrectly.